src/runtime_src/core/common/api/aie/xrt_graph.cpp

```diff
@@ -204,6 +225,110 @@ send_exception_message(const char* msg)
 
 }
 
+//////////////////////////////////////////////////////////////
+// xrt_graph C++ API implementations (xrt_graph.h)
+//////////////////////////////////////////////////////////////
+namespace xrt {
+
+graph::
+graph(const xrt::device& device, const xrt::uuid& xclbin_id, const std::string& name)
+  : handle(open_graph(device, xclbin_id, name))
+{}
+
+void
+graph::
+reset() const
+{
+  handle->reset();
+}
+
+uint64_t
+graph::
+get_timestamp() const
+{
+  return (handle->get_timestamp());
+}
+
+void
+graph::
+run()
+{
+  handle->run(0);
+}
+
+void
+graph::
+run(uint32_t iterations)
+{
+  handle->run(iterations);
+}
+
+void
+graph::
+wait_done(int timeout_ms)
+{
+  handle->wait(timeout_ms);
+}
+
+void
+graph::
+wait()
+{
+  handle->wait(0);
+}
+
+void
+graph::
+wait(uint64_t cycles)
```

Comment:
Add in the header
```suggestion
wait(std::uint64_t cycles = 0)
```
and remove the previous overload.

Reply:
Maybe just uint64_t and a wholesale change if <cstdint> can't be counted on to also put the types in global namespace, which all current implementations do.